### ISSUE TITLE
Swagger Body DTO not showing in documentation

### ISSUE THREAD

**Issue by Chowarmaan** [nestjs/nest#10281]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

I have created a very simple example of this issue, and it seems easily replicated.  The issue is I have created a DTO for a request and is not showing in the Swagger documentation, for the request.  Swagger has correctly identified the DTO (OneDTO), but does not include the information about it.  The response, which is the same DTO, is showing though, and the Schema section has the DTO information.  The attached image shows the issue.

<img width="321" alt="RequestDTONotShowing" src="https://user-images.githubusercontent.com/175738/190206616-69324db2-d81c-452f-9ac2-6e0027be85ac.png">

### Minimum reproduction code

https://github.com/Chowarmaan/IntersectTypeIssue

### Steps to reproduce

1. npm install
2. npm run start:dev
3. browse to Swagger docs (localhost:30000/api) and look at POST request body

### Expected behavior

The schema for the OneDTO should be populated in the Request section of the documentation, like the response and schema areas.

### Package

- [ ] I don't know. Or some 3rd-party package
- [ ] <code>@nestjs/common</code>
- [ ] <code>@nestjs/core</code>
- [ ] <code>@nestjs/microservices</code>
- [ ] <code>@nestjs/platform-express</code>
- [ ] <code>@nestjs/platform-fastify</code>
- [ ] <code>@nestjs/platform-socket.io</code>
- [ ] <code>@nestjs/platform-ws</code>
- [ ] <code>@nestjs/testing</code>
- [ ] <code>@nestjs/websockets</code>
- [X] Other (see below)

### Other package

@nestjs/swagger

### NestJS version

9.1.3

### Packages versions

```json
{
  "name": "intersect-type-issue",
  "version": "0.0.1",
  "description": "",
  "author": "",
  "private": true,
  "license": "UNLICENSED",
  "scripts": {
    "prebuild": "rimraf dist",
    "build": "nest build",
    "format": "prettier --write \"src/**/*.ts\" \"test/**/*.ts\"",
    "start": "nest start",
    "start:dev": "nest start --watch",
    "start:debug": "nest start --debug --watch",
    "start:prod": "node dist/main",
    "lint": "eslint \"{src,apps,libs,test}/**/*.ts\" --fix",
    "test": "jest",
    "test:watch": "jest --watch",
    "test:cov": "jest --coverage",
    "test:debug": "node --inspect-brk -r tsconfig-paths/register -r ts-node/register node_modules/.bin/jest --runInBand",
    "test:e2e": "jest --config ./test/jest-e2e.json"
  },
  "dependencies": {
    "@nestjs/class-transformer": "^0.4.0",
    "@nestjs/class-validator": "^0.13.4",
    "@nestjs/common": "^9.0.0",
    "@nestjs/core": "^9.0.0",
    "@nestjs/platform-express": "^9.0.0",
    "@nestjs/swagger": "^6.1.2",
    "reflect-metadata": "^0.1.13",
    "rimraf": "^3.0.2",
    "rxjs": "^7.2.0"
  },
  "devDependencies": {
    "@nestjs/cli": "^9.0.0",
    "@nestjs/schematics": "^9.0.0",
    "@nestjs/testing": "^9.0.0",
    "@types/express": "^4.17.13",
    "@types/jest": "28.1.8",
    "@types/node": "^16.0.0",
    "@types/supertest": "^2.0.11",
    "@typescript-eslint/eslint-plugin": "^5.0.0",
    "@typescript-eslint/parser": "^5.0.0",
    "eslint": "^8.0.1",
    "eslint-config-prettier": "^8.3.0",
    "eslint-plugin-prettier": "^4.0.0",
    "jest": "28.1.3",
    "prettier": "^2.3.2",
    "source-map-support": "^0.5.20",
    "supertest": "^6.1.3",
    "ts-jest": "28.0.8",
    "ts-loader": "^9.2.3",
    "ts-node": "^10.0.0",
    "tsconfig-paths": "4.1.0",
    "typescript": "^4.7.4"
  },
  "jest": {
    "moduleFileExtensions": [
      "js",
      "json",
      "ts"
    ],
    "rootDir": "src",
    "testRegex": ".*\\.spec\\.ts$",
    "transform": {
      "^.+\\.(t|j)s$": "ts-jest"
    },
    "collectCoverageFrom": [
      "**/*.(t|j)s"
    ],
    "coverageDirectory": "../coverage",
    "testEnvironment": "node"
  }
}

```

[System Information]
OS Version     : Windows 10
NodeJS Version : v18.5.0
NPM Version    : 8.12.1

[Nest CLI]
Nest CLI Version : 9.1.3

[Nest Platform Information]
class-transformer version : 0.4.0
platform-express version  : 9.0.11
class-validator version   : 0.13.4
schematics version        : 9.0.3
swagger version           : 6.1.2
testing version           : 9.0.11
common version            : 9.0.11
core version              : 9.0.11
cli version               : 9.1.3

### Node.js version

18.5.0

### In which operating systems have you tested?

- [ ] macOS
- [X] Windows
- [ ] Linux

**Comment by micalevisk** [comment]
![image](https://user-images.githubusercontent.com/13461315/190209748-5a378716-256f-4a49-aae9-139a4469c9f6.png)

your code works as expected. Try it out after `rm -rf dist`

**Comment by Chowarmaan** [comment]
I just tried as you suggested, but I do not get this.  I am mentioning about the layout in the Request side, not the schema at the bottom.  In the image, I circled this in red.

If you notice the documents, the Request has a reference to the OneDTO, but it does not include the field definition or schema in the request section, but it does include this in the response area, and the schema.  The image you have attached is from the bottom schema section, and even the Response section, but not the Request section.
